src/main/business/UserManager/UserController.js

```diff
@@ -141,6 +122,40 @@ class UserController {
         const user = await this.getUser(data.username);
         const updateOperations = [];
         if (data.userType) {
+            if (!Object.values(USER_TYPES).some(v => v === data.userType)){
+                throw new EMSError(ERROR_MSGS.INVALID_TYPE(data.userType), ERROR_CODES.INVALID_TYPE);
+            }
+            if (user.getUserType() === USER_TYPES.ADMIN){
```

Comment:
We have a function for this.

If I am not wrong it called validateType

Reply:
you mean `verifyType`. 
Yes you're right. Problem is the error it throws comes with a meaningless message...